tom_observations/utils.py

```diff
@@ -1,14 +1,79 @@
+from astropy import units, coordinates
 from astropy.time import Time
 from astroplan import Observer, FixedTarget, time_grid_from_range
 import numpy as np
+from scipy import interpolate as interp
 import logging
 
 from tom_observations import facility
 
+#Work-around necessary until the NOAO main service upgrades are complete.
+#Would be sufficient to update astropy to v4 which handles this internally. But Wes doesn't want to do this to his work environment yet.
+#In future remove the two following lines and require an up to date astropy
+from astropy.utils import iers
+iers.Conf.iers_auto_url.set('ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals2000A.all')
+
 logger = logging.getLogger(__name__)
 
+d2r = np.pi/180.0
+
+def get_radec_ephemeris(eph_json_single, start_time, end_time, interval, observing_facility, observing_site):
+    observing_facility_class = facility.get_service_class(observing_facility)
+    sites = observing_facility_class().get_observing_sites()
+    observer = None
+    for site_name in sites:
+        obs_site = sites[site_name]
+        if obs_site['sitecode'] == observing_site:
+
+            observer = coordinates.EarthLocation(lat=obs_site.get('latitude')*units.deg,
+                                                 lon=obs_site.get('longitude')*units.deg,
+                                                 height=obs_site.get('elevation')*units.m)
+    if observer is None:
+        #this condition occurs if the facility being requested isn't in the site list provided.
+        return (None,None,None,None,-1)
+    ra = []
+    dec = []
+    mjd = []
+    for i in range(len(eph_json_single)):
+        ra.append(float(eph_json_single[i]['R']))
+        dec.append(float(eph_json_single[i]['D']))
+        mjd.append(float(eph_json_single[i]['t']))
+    ra = np.array(ra)
+    dec = np.array(dec)
+    mjd = np.array(mjd)
+
+    fra = interp.interp1d(mjd,ra)
+    fdec = interp.interp1d(mjd,dec)
+
+    start = Time(start_time)
+    end = Time(end_time)
+
+
+    time_range = time_grid_from_range(time_range=[start, end], time_resolution=interval*units.hour)
+    tr_mjd = time_range.mjd
+    #tr_mjd += interval/(24.0*2.0)
+
+
+    airmasses = []
+    sun_alts = []
+    for i in range(len(tr_mjd)):
+        c = SkyCoord(fra(time_range[i].mjd), fdec(time_range[i].mjd), frame="icrs", unit="deg")
+        t = Time(tr_mjd[i],format='mjd')#-8*units.h
+        sun = coordinates.get_sun(t)
+        altaz = c.transform_to(AltAz(obstime=t,location=observer))
+        sun_altaz = sun.transform_to(AltAz(obstime=t,location=observer))
+        airmass = altaz.secz
+        airmasses.append(airmass)
+        sun_alts.append(sun_altaz.alt.value)
+    airmasses = np.array(airmasses)
+    sun_alts = np.array(sun_alts)
+
+    if np.min(tr_mjd)>=np.min(mjd) and np.max(tr_mjd)<=np.max(mjd):
+        return (tr_mjd,fra(tr_mjd),fdec(tr_mjd),airmasses,sun_alts)
+    else:
+        return (None,None,None,None,-2)
```

Comment:
A log of little formatting errors between 50 and 75. I don't know if they would matter for this pull request. At the CADC we follow PEP8 (which is the standard formatting guide) and use an IDE plugin (usually) that highlights any PEP8 formatting issues. I would think/hope project follows PEP8. Do you use an IDE for python? Check if it has a PEP8 or flake8 plugin to highlight formatting errors.

Reply:
I used atom editor because it is light weight and my laptop is old and heavy weight. But the identification of these errors seems to fail. So once I get the code up and working, I usually use pycodestyle as a check. Just failed to check this file! Derp.